irssi-1.1.0 make check link errors on Solaris 11.3

Hi,

make check doesn't work anymore on Solaris 11.3 (1.0.6 was fine).

```
$ xzcat irssi-1.1.0.tar.xz | pax -r
$ cd irssi-1.1.0
$ PATH=$(getconf PATH) ./configure PKG_CONFIG_PATH=/usr/lib/64/pkgconfig MAKE=gmake NM=/usr/gnu/bin/nm CPPFLAGS='-D_LARGEFILE64_SOURCE -D_XOPEN_SOURCE=600 -D__EXTENSIONS__=1 -D_XPG6' CC=cc CFLAGS='-m64 -Ui386 -U__i386 -g -O -fvisibility=hidden -KPIC -DPIC -std=c11' LDFLAGS='-m64 -zaslr=enable -znxstack=enable -znxheap=enable -M /usr/lib/ld/map.noexbss -M /usr/lib/ld/map.noexdata -M /usr/lib/ld/map.pagealign -Bdirect -zignore -zstrip-class=comment -ztype=pie' --prefix=/usr --disable-silent-rules --with-perl=no
$ gmake
...
$ gmake check
...
gmake[6]: Entering directory `/tmp/irssi-1.1.0/tests/fe-common/core'
...
libtool: link: cc -m64 -Ui386 -U__i386 -g -O -fvisibility=hidden -KPIC -DPIC -std=c11 -m64 -zaslr=enable -znxstack=enable -znxheap=enable -M /usr/lib/ld/map.noexbss -M /usr/lib/ld/map.noexdata -M /usr/lib/ld/map.pagealign -Bdirect -zignore -zstrip-class=comment -ztype=pie -o test-formats test_formats-test-formats.o  ../../../src/fe-common/core/libfe_common_core.a ../../../src/core/libcore.a ../../../src/lib-config/libirssi_config.a -L/usr/lib/amd64 -lgmodule-2.0 -lpthread -lglib-2.0 -lssl -lcrypto
Undefined                       first referenced
 symbol                             in file
__xnet_connect                      ../../../src/core/libcore.a(network.o)  (symbol belongs to implicit dependency /usr/lib/amd64/libsocket.so.1)
...
ld: fatal: symbol referencing errors
gmake[6]: *** [test-formats] Error 2
```

This one can be fixed by adding `-lsocket -lnsl` to `irssi-1.1.0/tests/fe-common/core/Makefile`.

There's another one though:

```
gmake[6]: Entering directory `/tmp/irssi-1.1.0/tests/irc/core'
libtool: link: cc -m64 -Ui386 -U__i386 -g -O -fvisibility=hidden -KPIC -DPIC -std=c11 -m64 -zaslr=enable -znxstack=enable -znxheap=enable -M /usr/lib/ld/map.noexbss -M /usr/lib/ld/map.noexdata -M /usr/lib/ld/map.pagealign -Bdirect -zignore -zstrip-class=comment -ztype=pie -o test-irc test_irc-test-irc.o  ../../../src/irc/core/libirc_core.a ../../../src/core/libcore.a ../../../src/lib-config/libirssi_config.a -L/usr/lib/amd64 -lgmodule-2.0 -lpthread -lglib-2.0 -lssl -lcrypto
Undefined                       first referenced
 symbol                             in file
__xnet_connect                      ../../../src/core/libcore.a(network.o)  (symbol belongs to implicit dependency /usr/lib/amd64/libsocket.so.1)
...
ld: fatal: symbol referencing errors
gmake[6]: *** [test-irc] Error 2
```

This one can be fixed too by adding `-lsocket -lnsl` to `irssi-1.1.0/tests/irc/core/Makefile` (test_irc_LDADD).

All tests pass after that.

--- a/tests/fe-common/core/Makefile.am
+++ b/tests/fe-common/core/Makefile.am
@@ -21,8 +21,7 @@ test_formats_LDADD = \
 	../../../src/fe-common/core/libfe_common_core.a \
 	../../../src/core/libcore.a \
 	../../../src/lib-config/libirssi_config.a \
-	@GLIB_LIBS@ \
-	@OPENSSL_LIBS@
+	@PROG_LIBS@
 
 test_formats_SOURCES = \
 	test-formats.c

--- a/tests/irc/core/Makefile.am
+++ b/tests/irc/core/Makefile.am
@@ -22,8 +22,7 @@ test_irc_LDADD = \
 	../../../src/irc/core/libirc_core.a \
 	../../../src/core/libcore.a \
 	../../../src/lib-config/libirssi_config.a \
-	@GLIB_LIBS@ \
-	@OPENSSL_LIBS@
+	@PROG_LIBS@
 
 test_irc_SOURCES = \
 	test-irc.c

